### DISCUSSION TITLE
using this with existing Grafana instance

### DISCUSSION THREAD

**Discussion by fuadar** [jasonacox/Powerwall-Dashboard#172] [Q&A] [answered]
So i'm already running teslamate and the Grafana instance that comes with it.
i have a full docker compose working it has anyone broken down the setup.sh so we can decouple and reuse and existing  Grafana instance. i am going to download the repo and see if i can do something like it. I am planning to run this on my synology nas where i have teslamate running already

**Comment by jasonacox** [comment, marked as answer]
You should be able to remove the 'grafana' section from [powerwall.yml](https://github.com/jasonacox/Powerwall-Dashboard/blob/main/powerwall.yml#L61-L78) and still run setup.sh (or manually follow the steps [here](https://github.com/jasonacox/Powerwall-Dashboard#option-2---manual-install)). 

When you set up your existing Grafana using [these steps](https://github.com/jasonacox/Powerwall-Dashboard#grafana-setup), and add your Influxdb datasource, you will use something like http://localhost:8086 (instead of http://influxdb:8086).  Otherwise, the rest of the settings should be ok.

Would love to hear your experience and any tip/tricks you discover. 🙏

**Reply by vincentvasquez** [reply]
Hi Jason!

Just wanted to let you know that I've been trying this today, a good amount 😅. So initially when I removed the Grafana section, and tried importing the dashboard to my existing Grafana instance from Teslamate, the Auto Provisioning part doesn't work.

I then choose InfluxDB, and try http://localhost8086 and then get 
Get "http://localhost:8086/query?db=powerwall&epoch=ms&q=SHOW+measurements": dial tcp [::1]:8086: connect: connection refused error reading influxDB

I did try reverting and adding the Grafana section back in. a brand new clean setup does work, but now on a separate Grafana instance, although when I went to Dashboard\browse and New\Import, after selecting the dashboard, the data sources for both PW and Sun/Moon were showing auto configured. I imported and everything looks good in the separate Grafana instance.

Any other ideas on what I can try to get just 1 Grafana instance?

**Reply by jasonacox** [reply]
You could try updating http://localhost:8086/ to have the IP address of the host.

One note: the setup.sh script is designed to be the "easy" auto-mode way to get everything working (including things like upgrade.sh). Your customization make it more of a manual ('hard" mode ? :) setup so be careful not to use the setup.sh or upgrade.sh scripts going forward.
